### PR TITLE
feat(937): Display child pipeline secrets

### PR DESCRIPTION
**Pull request by Filbird**
## Context
Child pipelines inherit secrets from their parent pipeline and can override them.

## Objective
The secrets view for child pipelines should show secrets inherited from their parent pipeline. These secrets should be differentiated and should not be allowed to be updated/deleted. 

Instead, these secrets can be overridden by creating a secret of the same name from the child pipeline. Conversely the secrets can be reverted back to their parent pipeline's values.

screen shot 2018-06-12 at 10 56 40 am

## Reference
https://github.com/screwdriver-cd/screwdriver/issues/937

### CODE REVIEW

**Comment by Filbird** [comment]
@tkyi Sounds reasonable! Where do you figure the best place to display that description is? Underneath `Secrets` or perhaps underneath the table?

**Comment by jithin1987** [comment]
@Filbird same place as we display current message. https://github.com/screwdriver-cd/ui/blob/0ffe5365253fd5f7167852563fd3df69280bb5b4/app/components/pipeline-secret-settings/template.hbs#L9  Replace it with something of the line "Secrets are inherited from [parent pipeline](Link to parent pipeline secrets)". You have the option to update a Secret with new value.

**Comment by Filbird** [comment]
@jithin1987  Updated 😄 
screen shot 2018-06-13 at 11 50 25 am

File: app/components/secret-view/component.js [hunk 1]
```diff
@@ -20,16 +45,27 @@ export default Component.extend({
     modifySecret() {
       const secret = this.get('secret');
 
-      if (this.get('buttonAction') === 'Delete') {
-        secret.destroyRecord();
-      } else {
+      if (this.get('buttonAction') === 'Delete'
+        || this.get('buttonAction') === 'Revert') {
+        return secret.destroyRecord().then(() => {
+          this.get('secrets').store.unloadRecord(secret);
```
Comment (minz1027) [inline app/components/secret-view/component.js]: Why we still need to unloadRecord after `destroyRecord`? based on the doc, after the destroy it should be gone both locally and backend?

Comment (Filbird) [reply app/components/secret-view/component.js]: I had thought so as well, but then I was running into an issue similar to : https://github.com/emberjs/data/issues/4972.

In our case, if we revert then override the same secret we get a duplicate id error.

Comment (minz1027) [reply app/components/secret-view/component.js]: ah, got it! thx for explaining!

**Pull request merged**